app/controllers/partners_controller.rb

```diff
@@ -102,8 +97,13 @@ def set_title
         'All Partners'
       end
   end
-  # This controller doesn't allow CRUD
-  # def partner_params
-  #   params.fetch(:partner, {})
-  # end
+
+  def neighbourhoods_from(partners, badge_zoom_level)
```

Comment:
This should be a class method in the Partner model as it is entirely a database operation